### ISSUE TITLE
pkg/admission/limitranger/admission.go:102: watch of *api.LimitRange ended with: 401: The event i...

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#5313]
The etcd error 401 should be typed on watch, properly handled by the list-watcher (not logged via util.HandleError)

https://app.getsentry.com/red-hat/openshift-3-stg/group/81719366/

```
*errors.errorString: pkg/admission/limitranger/admission.go:102: watch of *api.LimitRange ended with: 401: The event in requested index is outdated and cleared (the requested history has been cleared [5759081/5756368]) [5760080]
  File "/builddir/build/BUILD/openshift-git-525.eddc479/_build/src/github.com/openshift/origin/pkg/cmd/util/serviceability/sentry.go", line 61, in CaptureError
  File "/builddir/build/BUILD/openshift-git-525.eddc479/_build/src/github.com/openshift/origin/pkg/cmd/util/serviceability/panic.go", line 28, in CaptureError.fm
  File "/builddir/build/BUILD/openshift-git-525.eddc479/_thirdpartyhacks/src/github.com/GoogleCloudPlatform/kubernetes/pkg/util/util.go", line 82, in HandleError
  File "/builddir/build/BUILD/openshift-git-525.eddc479/_thirdpartyhacks/src/github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache/reflector.go", line 227, in listAndWatch
  File "/builddir/build/BUILD/openshift-git-525.eddc479/_thirdpartyhacks/src/github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache/reflector.go", line 138, in func.004
...
(4 additional frame(s) were not displayed)
```

**Comment by ncdc** [comment]
P1? Really?

**Comment by ncdc** [comment]
@smarterclayton what do you mean by "typed on watch"? And what do you consider proper handling of this?

**Comment by smarterclayton** [comment]
This is a known type of error that means "start listing".  We shouldn't
call handleError on it, we should just stop the list and start again.

**Comment by ncdc** [comment]
Unfortunately by the time the error bubbles up to the reflector, it's no longer strongly-typed. Instead, we get this:

```go
Object: &unversioned.Status{
  Status:  unversioned.StatusFailure,
  Message: err.Error(),
},
```

We could grep `Message` for this specific error text, which is obviously brittle.

Backwards compatibility issues aside, I don't think we can modify the info that is placed in the Object, because that has to be serializable to JSON, and afaik `error` types themselves are not.

We could send something else back instead of `unversioned.Status` when we get this type of error. That might be 1 viable option.

Thoughts?

**Comment by smarterclayton** [comment]
The error needs to be typed in etcd storage, then up to the rest api (in
the watch response), and all the way back.  Let's come back to this after
1.1.

**Comment by ncdc** [comment]
It is typed in etcd storage. Kube ends up swallowing it.

**Comment by ncdc** [comment]
Ok, I'm lowering this to P2.

**Comment by smarterclayton** [comment]
Mostly fixed by #16807
